test/mocks/genMocks.ts

```diff
@@ -0,0 +1,155 @@
+import "mocha";
+import {
+    BigNumber,
+    ContractTransaction,
+    ethers,
+    Signer,
+} from "ethers";
+import { Block, Provider } from "@ethersproject/providers";
+import { GenSale } from "../../src/contracts";
+
+export const feb23TsSeconds = 1677173138;
+export const feb24TsSeconds = 1677264314;
+
+export type GenMocks = {
+    signer: Partial<Signer>;
+    genSale: Partial<GenSale>;
+    currentSaleConfiguration: [
+        string,
+        BigNumber,
+        BigNumber,
+        BigNumber,
+        string,
+        BigNumber,
+        BigNumber,
+        BigNumber
+    ] & {
+        sellerWallet: string;
+        parentDomainId: BigNumber;
+        salePrice: BigNumber;
+        amountForSale: BigNumber;
+        mintlistMerkleRoot: string;
+        startingMetadataIndex: BigNumber;
+        folderGroupID: BigNumber;
+        transactionLimit: BigNumber;
+    };
+    currentSaleStarted: boolean;
+    currentAmountForSale: BigNumber;
+    currentTransactionLimit: BigNumber;
+    currentSalePrice: BigNumber;
+    currentDomainsSold: BigNumber;
+    currentSaleStartTimestamp: BigNumber;
+    currentLatestBlock: Partial<Block>;
+    currentSaleId: string;
+    currentSaleCounter: BigNumber;
+    currentPauseStatus: boolean;
+    currentSignerFunds: ethers.BigNumber;
+    currentDomainsPurchasedByAccount: BigNumber;
+    reset: () => void;
+};
+
+export function getDefaultMockGenSaleConfiguration():
+    [
+        string,
+        BigNumber,
+        BigNumber,
+        BigNumber,
+        string,
+        BigNumber,
+        BigNumber,
+        BigNumber
+    ] & {
+        sellerWallet: string;
+        parentDomainId: BigNumber;
+        salePrice: BigNumber;
+        amountForSale: BigNumber;
+        mintlistMerkleRoot: string;
+        startingMetadataIndex: BigNumber;
+        folderGroupID: BigNumber;
+        transactionLimit: BigNumber;
+    } {
```

Comment:
maybe extract into a subtype and reuse?

Reply:
Im not sure what you mean by that

Reply:
we have this type being used twice (in `mocks` object above as well). just a DRY nitpick :)

Reply:
Now this is using an interface type, so it should be resuable